Cleanup constants

moCello opened the pull request:
In this PR:
- Rename `DECOMPOSITION_MIN_ONE` to `V_I`
- Reverse order of `S_I` and `V_I`
- Add const functions for generating `S_I_BLS`, `DIVISOR`, `RECIP` and `S_BOX_BLS`
- Remove constants `BLS_ZERO` and `BLS_ONE` in favor of `const fn BlsScalar::zero()` and `const fn BlsScalar::one()` respectively
- Small fixes of comments and gitignore
- Add example for `zelbet` and `sponge` gadget

moCello committed:
Add examples for zelbet and sponge

The flamegraph add on for cargo is a great feature for performance
analysis and the added examples are the entry points for flamegraph.

moCello committed:
Rename constant to `V_I`

The new name is more consistent with the rest of the constants and more
descriptive than the old `DECOMPOSITION_MIN_ONE`.

Also have `witness_decomposition_gadget` use addition gates instead of
multiplication gates, which suits the use case more accurately.

moCello committed:
Reverse order of `S_I` and `V_I`

Resolves: #34

vlopes11 commented on src/constants.rs:
```diff
@@ -7,81 +7,178 @@
-/// BlsScalar one
-pub const BLS_ONE: Scalar = Scalar([
+/// BlsScalar with value 1
+const BLS_ONE: BlsScalar = BlsScalar([
```
Its unclear why do we need a constant for [BlsScalar::one](https://docs.rs/bls12_381/0.6.0/bls12_381/struct.Scalar.html#method.one)

moCello replied:
I use for the creation of the `MDS_MATRIX` constant a bit later. But if you prefer, I can also use `BlsScalar::one()` in the matrix.

vlopes11 commented on src/hash/zelbet/bar.rs:
```diff
@@ -4,47 +4,39 @@
+use dusk_plonk::prelude::BlsScalar;
 
 /// Convert representation from tuple in (Z_{s_n} x ... x Z_{s_1}) to single
 /// scalar element in Montgomery form (out of circuit)
```
Maybe unrelated to this PR, but its unclear what `scalar element in Montgomery form (out of circuit)` means here. Maybe rephrase this documentation?

moCello replied:
Good catch, fixed it in the following commits

moCello committed:
Remove flamegraph and performance files

moCello committed:
Remove constants `BLS_ZERO` and `BLS_ONE`

Additionally this commit adds functions to generate the constants
`S_I_BLS`, `DIVISORS`, `RECIP` and `S_BOX_BLS` at compile time.

Resolves: #39

vlopes11 reviewed (approved):
LGTM
